node/primitives/src/disputes/mod.rs

```diff
@@ -69,6 +72,99 @@ impl CandidateVotes {
 	}
 }
 
+#[derive(Debug, Clone)]
+/// Valid candidate votes.
+///
+/// Prefere backing votes over other votes.
+pub struct ValidCandidateVotes {
+	votes: BTreeMap<ValidatorIndex, (ValidDisputeStatementKind, ValidatorSignature)>,
+}
```

Comment:
nit: I admit I don't see why this extra indirection over the btreemap is necessary. Can `ValidCandidateVotes` be a `type` instead?

Reply:
If it were just a type and `insert_vote` its own function, it would be too easy to not use `insert_vote` but just a plain `insert`, which is really not desired. I usually try to create interfaces that make it easy to be used correctly and inconvenient/hard to be used incorrectly.

Reply:
Ah got it, thank you for explaining! I agree with the decision now. It's just of unfortunate that there's so much boiler-plate, and the abstraction is still leaky due to extensive use of `raw()` everywhere. I read over [this doc](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) but couldn't find a solution. Anyway, I'm being super nitty, it's good as-is.

Reply:
Yeah the `raw` is a bit ugly, but it is not leaky in the way I would care, as it only provides read only access. So it is not possible to bypass `insert_vote` via `raw()`.